<a href="https://colab.research.google.com/github/asalva15/Stage-ERC-2023/blob/main/T1_analyse_de_longueur.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
import os

In [ ]:
import json

In [ ]:
from google.colab import drive


In [ ]:
drive.mount('/content/drive')


In [ ]:
file_path_essai = '/content/drive/My Drive/ERC 2023 Axelle Salvador/essai.csv'


In [ ]:
os.chdir(os.path.dirname(file_path_essai))

In [ ]:
essai = pd.read_csv(file_path_essai, encoding="latin-1")
essai.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Phrase  5 non-null      object
dtypes: object(1)
memory usage: 168.0+ bytes


In [ ]:
print(essai)

                                              Phrase
0                                    Je vais manger.
1                 Qui rira bien qui rira le dernier.
2  Jean Anouilh est l'auteur de la pièce Le Voyag...
3  On est quelquefois aussi différent de soi-même...
4  Rien n'empêche tant d'être naturel que l'envie...


In [ ]:
dic_essai= {}

In [ ]:

for index, row in essai.iterrows():
    sentence = row['Phrase']
    words = sentence.split() #par défaut, splt sur l'espace entre les mots
    length = len(words)
    dic_essai[sentence] = length

print(sentence_length_dict) #à revoir, notamment syntaxe du nom


{'Phrase': 1, 'Je vais manger.': 3, 'Qui rira bien qui rira le dernier.': 7, "Jean Anouilh est l'auteur de la pièce Le Voyageur sans bagages. ": 11, 'On est quelquefois aussi différent de soi-même que des autres. ': 10, "Rien n'empêche tant d'être naturel que l'envie de le paraître.": 10}


In [ ]:
with open(json_file_path, 'w',encoding="utf-8") as json_file:
    json.dump(sentence_length_dict, json_file, indent=4)

In [ ]:
file_path = '/content/drive/My Drive/ERC 2023 Axelle Salvador/modern_all_alignment.csv'

In [ ]:
os.chdir(os.path.dirname(file_path))

In [ ]:
import sys

csv.field_size_limit(sys.maxsize)

9223372036854775807

In [ ]:
modern_all_alignment = pd.read_csv(file_path, encoding="latin-1",engine="python")
modern_all_alignment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1878199 entries, 0 to 1878198
Data columns (total 10 columns):
 #   Column           Dtype  
---  ------           -----  
 0   source_passage   object 
 1   source_author    object 
 2   source_title     object 
 3   source_date      float64
 4   target_passage   object 
 5   target_author    object 
 6   target_title     object 
 7   target_date      float64
 8   source_filename  object 
 9   target_filename  object 
dtypes: float64(2), object(8)
memory usage: 143.3+ MB


In [ ]:
dic_all= {}
modern_all_alignment.dropna(subset=['source_passage', 'target_passage'], inplace=True)
list_all = modern_all_alignment['source_passage'] + modern_all_alignment['target_passage']
nb_words = [len(passage.split()) for passage in list_all]
df = pd.DataFrame.from_dict({'passage': list_all, 'nb_words': nb_words})

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1878066 entries, 0 to 1878198
Data columns (total 2 columns):
 #   Column    Dtype 
---  ------    ----- 
 0   passage   object
 1   nb_words  int64 
dtypes: int64(1), object(1)
memory usage: 43.0+ MB


In [248]:
df.agg(
    {
        "nb_words": ["min","max","median"]
    }
)

nb_words
min          1.0
max     140993.0
median      63.0

In [237]:
df[df['nb_words'] == 1]['passage']

121878     http://gallica.bnf.fr/ark:/12148/bpt6k6516949w...
121879     reconnoissancesp="http://gallica.bnf.fr/ark:/1...
121880     reconnoissancesp="http://gallica.bnf.fr/ark:/1...
121881     http://gallica.bnf.fr/ark:/12148/bpt6k6516949w...
121882     reconnoissancesp="http://gallica.bnf.fr/ark:/1...
                                 ...                        
1790234    http://gallica.bnf.fr/ark:/12148/bpt6k5785741s...
1790235    connoissancesp="http://gallica.bnf.fr/ark:/121...
1790237    connoissancesp="http://gallica.bnf.fr/ark:/121...
1860380    http://gallica.bnf.fr/ark:/12148/bpt6k65600536...
1860382    http://gallica.bnf.fr/ark:/12148/bpt6k65600536...
Name: passage, Length: 342, dtype: object

In [246]:
q3 = df['nb_words'].quantile(0.75)
print(q3)


140.0


Le seuil choisi correspond au troisième quartile : 140 mots.